src/logic/config/utils/index.ts

```diff
@@ -2,8 +2,10 @@ import { _getChainId, _setChainId } from 'src/config'
 import { ChainId } from 'src/config/chain.d'
 import { store } from 'src/store'
 import { setChainIdAction } from 'src/logic/config/store/actions'
+import { instantiateSafeContracts } from 'src/logic/contracts/safeContracts'
 
 export const setChainId = (newChainId: ChainId) => {
   _setChainId(newChainId)
   store.dispatch(setChainIdAction(newChainId))
+  instantiateSafeContracts()
```

Comment:
A job for a middleware?

Reply:
I mean, this is using our local `_chainId`, not the one from the store. It could be moved to a middleware (as well as `_setChainId` actually). What do you think?

Reply:
It's doing `store.dispatch` so I've assumed it will trigger a store update.